### PR TITLE
Support class and realname props for a folder

### PR DESCRIPTION
**Pull request by tpayen**
Hooks like `render_mailboxlist` allow a plugin to customize the folder tree but are limited for the folder css class and the folder display name. 

With this PR a plugin can add a `class` prop and a `realname` boolean prop to customize the CSS class of the folder and the display name.

### CODE REVIEW

**Review by alecpl — CHANGES_REQUESTED** [review]

**Comment by tpayen** [comment]
Hi @alecpl thanks for your response. I took your comments into account in my last commit

File: program/include/rcmail_action.php [hunk 1]
```diff
@@ -1325,7 +1326,7 @@ protected static function render_folder_tree_select(&$arrFolders, &$mbox_name, $
                 }
             }
 
-            if (!$realnames && ($folder_class = self::folder_classname($folder['id'])) && $rcmail->text_exists($folder_class)) {
+            if (!$realnames && ($folder_class = self::folder_classname($folder['id'], isset($folder['class']) ? $folder['class'] : null)) && $rcmail->text_exists($folder_class)) {
```
Comment (alecpl) [inline program/include/rcmail_action.php:1329]: This line is too long now.

File: program/include/rcmail_action.php [hunk 2]
```diff
@@ -1353,10 +1354,11 @@ protected static function render_folder_tree_select(&$arrFolders, &$mbox_name, $
      * (including shared/other users namespace roots).
      *
      * @param string $folder_id IMAP Folder name
+     * @param string $fallback fallback Folder CSS class name
```
Comment (alecpl) [inline program/include/rcmail_action.php:1357]: Two spaces after $fallback, and start the description with a capital letter.

File: program/include/rcmail_action.php [hunk 3]
```diff
@@ -1216,15 +1216,16 @@ protected static function render_folder_tree_html(&$arrFolders, &$mbox_name, &$j
         $out = '';
         foreach ($arrFolders as $folder) {
             $title        = null;
-            $folder_class = self::folder_classname($folder['id']);
+            $folder_class = self::folder_classname($folder['id'], isset($folder['class']) ? $folder['class'] : null);
             $is_collapsed = strpos($collapsed, '&'.rawurlencode($folder['id']).'&') !== false;
             $unread       = 0;
+            $realname     = $realnames || isset($folder['realname']) ? $folder['realname'] : false;
```
Comment (alecpl) [inline program/include/rcmail_action.php:1222]: Here you need to add brackets. Or, it should be:
```
$realname     = isset($folder['realname']) ? $folder['realname'] : $realnames;
```

**Review by alecpl — CHANGES_REQUESTED** [review]

File: program/include/rcmail_action.php [hunk 4]
```diff
@@ -1216,15 +1216,16 @@ protected static function render_folder_tree_html(&$arrFolders, &$mbox_name, &$j
...
             $is_collapsed = strpos($collapsed, '&'.rawurlencode($folder['id']).'&') !== false;
             $unread       = 0;
+            $realname     = isset($folder['realname']) ? $folder['realname'] : $realnames;
 
             if ($msgcounts && !empty($msgcounts[$folder['id']]['UNSEEN'])) {
                 $unread = intval($msgcounts[$folder['id']]['UNSEEN']);
             }
 
-            if ($folder_class && !$realnames && $rcmail->text_exists($folder_class)) {
+            if (isset($folder_class) && !$realname && $rcmail->text_exists($folder_class)) {
```
Comment (alecpl) [inline program/include/rcmail_action.php:1228]: Here we don't need isset().

File: program/include/rcmail_action.php [hunk 5]
```diff
@@ -1325,7 +1326,10 @@ protected static function render_folder_tree_select(&$arrFolders, &$mbox_name, $
                 }
             }
 
-            if (!$realnames && ($folder_class = self::folder_classname($folder['id'])) && $rcmail->text_exists($folder_class)) {
+            $folder_class = self::folder_classname($folder['id'], isset($folder['class']) ? $folder['class'] : null);
+            $realname     = isset($folder['realname']) ? $folder['realname'] : $realnames;
+
+            if (isset($folder_class) && !$realname && $rcmail->text_exists($folder_class)) {
```
Comment (alecpl) [inline program/include/rcmail_action.php:1332]: Here we don't need isset().

**Review by alecpl — CHANGES_REQUESTED** [review]

**Pull request merged**